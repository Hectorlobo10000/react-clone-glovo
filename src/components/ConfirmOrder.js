/** @jsx jsx */
import { jsx } from "@emotion/core";
import { Button } from "../components/ui";
import { usePostOrder } from "../action-hook";
import { useRestId, useSelectorCart } from "../selector";
import { navigate } from "@reach/router";
import { useReset } from "../action-hook";

function ConfirmOrder() {
  const order = usePostOrder();
  const restid = useRestId();
  const cart = useSelectorCart();
  const reset = useReset();

  function sendOrder() {
    const data = {
      order: {
        restaurant_id: restid,
        order_items_attributes: Object.values(cart).map(value => {
          return {
            menu_item_id: value.id,
            quantity: value.quantity
          };
        })
      }
    };
    order(data);
    reset();
    navigate("/");
  }
  return (
    <div>
      <Button
        css={{
          backgroundColor: "rgba(255,210,0,1)",
          color: "white",
          "&:hover": {
            backgroundColor: "rgba(255,210,0,0.9)",
            color: "white"
          }
        }}
        onClick={sendOrder}
      >
        Confirm Order
      </Button>
    </div>
  );
}

export default ConfirmOrder;
