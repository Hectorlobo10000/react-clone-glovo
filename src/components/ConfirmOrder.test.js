import React from "react";
import { render } from "@testing-library/react";
import ConfirmOrder from "./ConfirmOrder";
test("Confirm Order Form on Submit", () => {
  const { asFragment } = render(<ConfirmOrder />);
  expect(asFragment()).toMatchSnapshot();
});
