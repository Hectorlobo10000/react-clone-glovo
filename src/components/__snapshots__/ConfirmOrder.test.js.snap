// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm Order Form on Submit 1`] = `
<DocumentFragment>
  .emotion-0 {
  background-color: rgba(0,168,133,1);
  border: none;
  border-radius: .25rem;
  color: white;
  cursor: pointer;
  font-size: 1.2rem;
  padding: .75rem 0;
  -webkit-transition: all 200ms ease;
  transition: all 200ms ease;
  text-align: center;
  font-weight: bold;
  text-transform: uppercase;
  width: 100%;
  background-color: rgba(255,210,0,1);
  color: white;
}

.emotion-0:hover {
  background-color: rgba(0,168,133,0.8);
  color: white;
}

.emotion-0:hover {
  background-color: rgba(255,210,0,0.9);
  color: white;
}

<div>
    <button
      class="emotion-0"
    >
      Confirm Order
    </button>
  </div>
</DocumentFragment>
`;
